### DISCUSSION TITLE
Solved: Attempting to map a list of strings to #action

### DISCUSSION THREAD

**Discussion by shmup** [scandum/tintin#25] [Script help]
Attempting something like this, but sadly the `$msg` isn't expanded, and is treated as a literal.

```
#foreach {
  {You cease your meditation having achieved full will and mana.}
  {Your mental prowess is already swollen to its maximum.}
  {You open your eyes and stretch languidly, feeling deliciously well-rested.}
  {The sharp bite of peppery, clove-scented perfume freshens the air around you.}
} {msg} {
  #act $msg alert;
}
```

```
################################### ACTIONS ####################################
#ACTION {$msg} {alert}
```

Would love some help, though I may find the answer.

Note that I can `#echo $msg` just fine, in the foreach body.

**Comment by shmup** [comment]
On further inspection, it is a delicious bug I feel, ha!

Note that the last thing in the list _does_ register as an actual action, like, that string will trigger my alert, tho that is not obvious from the list when just viewing `#act`. Most strange..

**Reply by scandum** [reply]
Easiest solution is to force a variable substitution, which you can do as following:

```
#foreach {
  {You cease your meditation having achieved full will and mana.}
  {Your mental prowess is already swollen to its maximum.}
  {You open your eyes and stretch languidly, feeling deliciously well-rested.}
  {The sharp bite of peppery, clove-scented perfume freshens the air around you.}
} {msg} {
  #line substitute variables #act {$msg} {alert};
}
```

**Reply by shmup** [reply]
I definitely misunderstood #line, as I was trying to use it within {$msg} itself, and not on the entire uh, line.

Thanks @scandum, that squashes this and **all works perfectly.**

Though may be worth creating a ticket to track this other bug, where it technically was registering an `#act` to the last string in the list. The bug is just minor and in the output of `#act` with the literal `$msg`.

re: output below, and knowing that the alert is still firing for "The sharp bite of peppery, clove-scented perfume freshens the air around you.", and only that one.

```
################################### ACTIONS ####################################
#ACTION {$msg} {alert}
```

**Reply by scandum** [reply]
That's expected behavior. TinTin substitutes $msg right before the action is checked, though it's not very efficient. So it will trigger to whatever $msg is set to, which should be {The sharp bite of peppery, clove-scented perfume freshens the air around you.}
